File: Moodle/Budega/Mercantil.java
import java.util.*;

public class Mercantil 
{
    private ArrayList<Pessoa> caixas = new ArrayList<Pessoa>();
    private ArrayList<Pessoa> esperando = new ArrayList<Pessoa>();

    private boolean validarIndice(int index)
    {
        if (index >= 0)
        {
            return true;
        }

        return false;
    }

    Mercantil(int qtdCaixas)
    {
        if (!validarIndice(qtdCaixas))
        {
            return;
        }
        
        for (int i = 0; i < qtdCaixas; i++)
        {
            caixas.add(0, new Pessoa("-----"));
        }
    }

    public void chegar(Pessoa pessoa)
    {
        esperando.add(pessoa);
    }

    public void chamar(int index)
    {
        if (!validarIndice(index))
        {
            return;
        }

        try 
        {
            esperando.get(0);
        } 
        
        catch (IndexOutOfBoundsException e) 
        {
            System.out.println("fail: sem clientes");
            
            return;
        }

        if (caixas.get(index).getNome() != "-----")
        {
            System.out.println("fail: caixa ocupado");
            
            return;
        }

        caixas.set(index, esperando.get(0));
        esperando.remove(0);
    }

    public void finalizar(int index)
    {
        if (!validarIndice(index))
        {
            return;
        }

        try 
        {
            if (caixas.get(index).getNome() == "-----")
            {
                System.out.println("fail: caixa vazio");
            }
            
            caixas.set(index, new Pessoa("-----"));
        } 
        
        catch (IndexOutOfBoundsException e) 
        {
            System.out.println("fail: caixa inexistente");
        }
    }

    public String show()
    {
        return "Caixas: [" + showFor(caixas) + "]\nEspera: [" + showFor(esperando) + "]";
    }

    public String showFor(ArrayList<Pessoa> pessoa)
    {
        String list = "";

        for (Pessoa i : pessoa)
        {
            list += i.show();
            if (!pessoa.get(pessoa.size() - 1).equals(i))
            {
                list += ", ";
            }
        }

        return list;
    }
}

File: Extra/Budega (.py)/mercantil.py
from pessoa import Pessoa


class Mercantil():
    def __init__(self, qtdCaixas : int):
        self.caixas = [Pessoa("-----")] * qtdCaixas
        self.espera = []


    def __validarIndice(self, indice : int):
        if indice < len(self.caixas):
            return True
        
        raise Exception("fail: indice invalido")
    

    def chegar(self, pessoa : Pessoa):
        self.espera.append(pessoa)


    def chamar(self, indice : int):
        if self.__validarIndice(indice):
            self.caixas.remove(self.caixas[indice])
            self.caixas.insert(indice, self.espera[0])

            self.espera.remove(self.espera[0])


    def finalizar(self, indice : int):
        if not self.__validarIndice(indice):
            return
        

        if self.caixas[indice] == Pessoa("-----"):
            raise Exception("fail: caixa vazio")
        
        self.caixas.remove(self.caixas[indice])
        self.caixas.insert(indice, Pessoa("-----"))

    
    def __str__(self):
        return f"Caixas: {self.caixas}\nEspera: {self.espera}"
    
    
    def __repr__(self):
        return f"Caixas: {self.caixas}\nEspera: {self.espera}"